middle_end/flambda2/to_cmm/to_cmm_static.ml

```diff
@@ -164,6 +164,19 @@ let static_const0 env r ~updates (bound_static : Bound_static.Pattern.t)
     in
     let env, e = static_float_array_updates s env updates 0 fields in
     env, R.update_data r float_array, e
+  | Block_like s, Immutable_value_array fields ->
+    let block_name = Symbol.linkage_name_as_string s, Cmmgen_state.Global in
+    let header = C.black_block_header 0 (List.length fields) in
+    let env, static_fields =
+      List.fold_right
+        (fun v (env, static_fields) ->
+          let static_field = static_value v in
+          env, static_field :: static_fields)
+        fields (env, [])
```

Comment:
The environment doesn't really need to be part of the accumulator (this also applies to the `Block` case, which is likely where this code comes from)